packages/nimble-components/src/rich-text/models/tests/markdown-serializer.spec.ts

```diff
@@ -56,6 +56,46 @@ describe('Markdown serializer', () => {
                 html: '<strong><em>Bold and Italics</em></strong>',
                 markdown: '***Bold and Italics***'
             },
+            {
+                name: 'Italics without spaces in between bold texts',
+                html: '<strong>Bold<em>italics</em>bold</strong>',
```

Comment:
As discussed in the following comment I have significant concerns with the usefulness of these tests: https://github.com/ni/nimble/pull/1496#discussion_r1324888037

As described in the PR description "Please note that the input HTML string given to the tests is the same as how the Tiptap renders bold and italics cases. " 

My concern is that these tests are manually created html strings and there is nothing to assert that when we update tiptap, etc that the HTML that would actually be generated won't change. The tests only reflect the current exact version of the dependencies used and I do not think are significantly useful for testing potential regressions or change in behavior.

I won't block this PR but we need to discuss an different testing strategy.

Reply:
Yes, we agree with your concerns whatever is posted here and in the other PR comment regarding the testing strategy followed for serializer logic. I have created a tech debt issue https://github.com/ni/nimble/issues/1533 to update the testing approach but we will try to take it immediately after this PR.

I am keeping these test cases as it is for now because these tests will check the fix for `expelEnclosingWhitespace` in the ProseMirror and might be captured if it breaks in future releases of ProseMirror. Also, when we update the testing approach, we will revisit the cases and add exactly what is required.